bonfire/qontract.py

```diff
@@ -146,19 +146,28 @@ def _find_matching_component(apps, app_name, component_name):
 
 
 def _check_replace_other(other_params, this_params):
+    # prefer stable frontend targets over beta targets
+    this_frontend_stable = this_params.get("ENV_NAME") != "frontends-beta"
+    other_frontend_stable = other_params.get("ENV_NAME") != "frontends-beta"
+    if this_frontend_stable and not other_frontend_stable:
+        return True
```

Comment:
Negative conditionals and double negatives confuse me, could we recast this like:

```
def _is_stable_frontend(params):
    return params.get("ENV_NAME") == "frontends"
    
def _check_replace_other(other_params, this_params):
    this_frontend_stable = _is_stable_frontend(this_params)
    other_frontend_stable = _is_stable_frontend(other_params)
    
    if this_frontend_stable and not other_frontend_stable:
        return True
```

Reply:
Check it out now, renamed it to `def _is_preferred(params)`

Reply:
I think it opens up the door to being able to customize what is preferred or not in future. Like for example we could have a CLI argument for `--prefer ENV_NAME=frontends-beta` ... we could even allow you to prefer multiple parameters which are OR'd or AND'd together.